crates/bevy_sprite/src/render/mod.rs

```diff
@@ -771,7 +776,7 @@ pub fn prepare_sprite_image_bind_groups(
             transparent_phase.items[batch_item_index]
                 .batch_range_mut()
                 .end += 1;
-            batches.last_mut().unwrap().1.range.end += 1;
+            batches.get_mut(&batch_key.unwrap()).unwrap().range.end += 1;
```

Comment:
Could add a `current_batch` variable containing a reference to the current `SpriteBatch` in the sprite batches map and avoid the extra hashmap look up here each iteration.

Reply:
This was a substantial improvement, now compared to main:

![image](https://github.com/user-attachments/assets/f415a316-41d6-47f3-938b-9c85c7a98fc8)